pkg/sql/schemachanger/scplan/internal/scstage/build.go

```diff
@@ -553,13 +678,30 @@ func (sb stageBuilder) String() string {
 // may prove to be a somewhat common pattern in other cases: consider the
 // intermediate index needed when adding and dropping columns as part of the
 // same transaction.
-func (bc buildContext) computeExtraJobOps(cur, next *Stage) []scop.Op {
-	// Schema change job operations only affect mutation stages no sooner
+func (bc buildContext) computeExtraOps(cur, next *Stage) []scop.Op {
+	// Schema change extra operations only affect mutation stages no sooner
 	// than pre-commit.
-	if cur.Phase < scop.PreCommitPhase || cur.Type() != scop.MutationType {
+	if cur.Type() != scop.MutationType {
+		return nil
+	}
+	var initializeSchemaChangeJob bool
+	switch cur.Phase {
+	case scop.StatementPhase:
 		return nil
+	case scop.PreCommitPhase:
+		if next == nil {
+			// This is the main stage, followed by no post-commit stages, do nothing.
```

Comment:
I think that this is the case where we should return an explicit operation to flush the progress and then be done.

Reply:
Can we leave that for later? I've been going back and forth on explicit flush ops, there seem to be many ways to skin a cat. I propose we sit on it for a couple of weeks and by then we'll be wiser about what the EXPLAIN (DDL) diagrams should look like in this new paradigm.